src/edu/csus/ecs/pc2/ui/ShadowCompareRunsPane.java

```diff
@@ -303,10 +361,372 @@ public void actionPerformed(ActionEvent arg0) {
+            //get the next submission id
+            //TODO: use an Enum for columnIds instead of hard-coded integers, which can change if the table is rearranged
+            Integer nextSubmissionId = (Integer) resultsTable.getModel().getValueAt(modelRow, 3); // 3=SubmissionID
+
+            //add the submissionID to the map of submissions to be updated, attaching its model row number
+            submissionIdMap.put(nextSubmissionId, modelRow);
+            
+            //debug:
+//            System.out.println ("   Submision ID: " + nextSubmissionId);
+            
+        }
+        
+        int numRunsUpdated = 0;
+        for (int submissionId : submissionIdMap.keySet()) {
+                        
+            //make sure the run is not "Pending" in either system
+            Integer modelRow = submissionIdMap.get(submissionId);
+            String pc2Judgement = (String) resultsTable.getModel().getValueAt(modelRow, 4);      // 4 = PC2 Judgment
+            String remoteJudgement = (String) resultsTable.getModel().getValueAt(modelRow, 5);   // 5=remote CCS judgement
+            
+            if (pc2Judgement.toLowerCase().contains("pending") || remoteJudgement.toLowerCase().contains("pending")) {
+                
+                JOptionPane.showMessageDialog(this, "Cannot update submission " + submissionId + " because it still has judgements Pending", 
+                                                    "Submission judgement Pending", JOptionPane.WARNING_MESSAGE);
+                log.log (Log.INFO, "Attempted to update pending submission: " + submissionId);
+                
+            } else {
+            
+                //verify the remote judgement is a valid value
+                boolean isClicsAcronym = CLICSJudgementType.isCLICSAcronym(remoteJudgement);
+                if (isClicsAcronym) {
+                    
+                    //get the CLICS acronym for the specified judgement 
+                    CLICS_JUDGEMENT_ACRONYM judgementAcronym = CLICSJudgementType.getCLICSAcronymFromElementName(remoteJudgement);
+
+                    // run is not pending in either system and the remote judgement is a valid CLICS judgement; 
+                    // attempt to update PC2 to match remote CCS judgement
+                    boolean success = updateRun(submissionId, judgementAcronym);
+                    
+                    if (success) {
+                        numRunsUpdated++;
+                    }
+                    
+                } else {
+                    
+                    //there is no known judgement matching the judgement string
+                    JOptionPane.showMessageDialog(this, "Cannot update submission " + submissionId + ": invalid remote judgement: " + remoteJudgement, 
+                                                        "Invalid judgement acronym", JOptionPane.WARNING_MESSAGE);
+                    log.log (Log.INFO, "Attempted to update pending submission: " + submissionId);
```

Comment:
This log message does not contain enough context Update to: 
"Attempted to update pending submission: " + submissionId) + 
" Invalid/Non-CLICS remote judgment acronym " + remoteJudgement